sdv/data_processing/data_processor.py

```diff
@@ -323,138 +323,433 @@ def create_regex_generator(self, column_name, sdtype, column_metadata, is_numeri
+                A transformer.
+
+        Returns:
+            rdt.transformers.Transformer:
+                A transformer.
+        """
+        if kwargs:
+            transformer_class = transformer.__class__
+            default_transformer_kwargs = _get_transformer_init_kwargs(transformer)
+            return transformer_class(**{**default_transformer_kwargs, **kwargs})
+
+        return deepcopy(transformer)
+
+    def _get_datetime_transformer(self, kwargs, transformer):
+        """Get a datetime transformer.
+
+        Args:
+            kwargs (dict):
+                The keyword arguments for the transformer.
+            transformer (rdt.transformers.DatetimeFormatter):
+                A datetime transformer.
+
+        Returns:
+            rdt.transformers.DatetimeFormatter:
+                A datetime transformer.
+        """
+        kwargs['enforce_min_max_values'] = self._enforce_min_max_values
+
+        return self._get_other_transformer(kwargs, transformer)
+
+    def _get_categorical_transformer(self, kwargs, transformer):
+        """Get a categorical transformer.
+
+        Args:
+            kwargs (dict):
+                The keyword arguments for the transformer.
+            transformer:
+                A categorical transformer.
+
+        Returns:
+            transformer:
+                A categorical transformer.
+        """
+        return self._get_other_transformer(kwargs, transformer)
+
+    def _get_numerical_transformer(self, kwargs, transformer):
+        """Get a numerical transformer.
+
+        Args:
+            kwargs (dict):
+                The keyword arguments for the transformer.
+            transformer:
+                A numerical transformer.
+
+        Returns:
+            transformer:
+                A numerical transformer.
+        """
+        return self._get_other_transformer(kwargs, transformer)
+
+    def _get_boolean_transformer(self, kwargs, transformer):
+        """Get a boolean transformer.
+
+        Args:
+            kwargs (dict):
+                The keyword arguments for the transformer.
+            transformer:
+                A boolean transformer.
+
+        Returns:
+            transformer:
+                A boolean transformer.
+        """
+        return self._get_other_transformer(kwargs, transformer)
+
+    def _get_sdtype_transformer(self, sdtype, kwargs, transformer):
+        transformer_map = {
+            'datetime': self._get_datetime_transformer,
+            'numerical': self._get_numerical_transformer,
+            'categorical': self._get_categorical_transformer,
+            'boolean': self._get_boolean_transformer,
+        }
+        transformer_fn = transformer_map.get(sdtype)
+        if transformer_fn:
+            return transformer_fn(kwargs, transformer)
+
+        return self._get_other_transformer(kwargs, transformer)
+
     def _get_transformer_instance(self, sdtype, column_metadata):
```

Comment:
This can be renamed to `_get_transformer_for_sdtype` or something similar to be coherent with the other suggested change.